feature/gribi/ate_tests/base_hierarchical_nhg_update/base_hierarchical_nhg_update_test.go

```diff
@@ -125,15 +132,67 @@ func TestBaseHierarchicalNHGUpdate(t *testing.T) {
+	validateTrafficFlows(t, ate, []*ondatra.Flow{p2flow}, []*ondatra.Flow{p3flow}, nil, pMACFilter)
+
+	t.Logf("Adding a new NH via port %v with ID %v", dutP3, p3NHID)
+	addNH(ctx, t, gribic, p3NHID, dutP3, pMAC)
+	t.Logf("Performing implicit in-place replace with two next-hops (NH IDs: %v and %v)", p2NHID, p3NHID)
+	addNHG(ctx, t, gribic, interfaceNHGID, []uint64{p2NHID, p3NHID})
+	validateTrafficFlows(t, ate, nil, nil, []*ondatra.Flow{p2flow, p3flow}, pMACFilter)
+	t.Logf("Performing implicit in-place replace using the next-hop with ID %v", p3NHID)
+	addNHG(ctx, t, gribic, interfaceNHGID, []uint64{p3NHID})
+	validateTrafficFlows(t, ate, []*ondatra.Flow{p3flow}, []*ondatra.Flow{p2flow}, nil, pMACFilter)
+	t.Logf("Performing implicit in-place replace using the next-hop with ID %v", p2NHID)
+	addNHG(ctx, t, gribic, interfaceNHGID, []uint64{p2NHID})
+	validateTrafficFlows(t, ate, []*ondatra.Flow{p2flow}, []*ondatra.Flow{p3flow}, nil, pMACFilter)
+}
 
-	validateTrafficFlows(t, ate, []*ondatra.Flow{p2flow}, []*ondatra.Flow{p3flow})
-
-	addInterfaceRoute(ctx, t, gribic, p3ID, dut.Port(t, "port3").Name(), atePort3.IPv4)
+// addNH adds a GRIBI NH with a FIB ACK confirmation via Modify RPC
+func addNH(ctx context.Context, t *testing.T, gribic *fluent.GRIBIClient, id uint64, intf, mac string) {
+	nh := fluent.NextHopEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithIndex(id).WithInterfaceRef(intf).WithMacAddress(mac)
+	gribic.Modify().AddEntry(t, nh)
+	if err := awaitTimeout(ctx, gribic, t, time.Minute); err != nil {
+		t.Fatalf("Await got error for entries: %v", err)
+	}
+	wantOperationResults := []*client.OpResult{
+		fluent.OperationResult().
+			WithNextHopOperation(id).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+	}
+	for _, wantResult := range wantOperationResults {
+		chk.HasResult(t, gribic.Results(t), wantResult, chk.IgnoreOperationID())
+	}
+}
 
-	validateTrafficFlows(t, ate, []*ondatra.Flow{p3flow}, []*ondatra.Flow{p2flow})
+// addNHG adds a GRIBI NHG with a FIB ACK confirmation via Modify RPC
+func addNHG(ctx context.Context, t *testing.T, gribic *fluent.GRIBIClient, id uint64, nhs []uint64) {
+	nhg := fluent.NextHopGroupEntry().WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithID(id)
+	for _, nh := range nhs {
+		nhg.AddNextHop(nh, 1)
+	}
+	gribic.Modify().AddEntry(t, nhg)
+	if err := awaitTimeout(ctx, gribic, t, time.Minute); err != nil {
+		t.Fatalf("Await got error for entries: %v", err)
+	}
+	wantOperationResults := []*client.OpResult{
+		fluent.OperationResult().
+			WithNextHopGroupOperation(id).
+			WithProgrammingResult(fluent.InstalledInFIB).
+			WithOperationType(constants.Add).
+			AsResult(),
+	}
+	for _, wantResult := range wantOperationResults {
+		chk.HasResult(t, gribic.Results(t), wantResult, chk.IgnoreOperationID())
+	}
 }
 
 // addDestinationRoute creates a GRIBI route to dstPfx via interfaceNH.
```

Comment:
Can you help update the comment to be more aligned? `blablabla to dstPfx via a VIP`.